src/lambda/commands/createNewSamApp.ts

```diff
@@ -42,50 +50,89 @@ export async function resumeCreateNewSamApp(context: Pick<vscode.ExtensionContex
 }
 
 interface NewSamAppMetadata {
+    reason: string
```

Comment:
Would it make sense to turn this into an enum?

Reply:
If an error is caught, the thought is to place the error message into reason. The rest of the uses could be considered as enums though.

Reply:
In that case, would it make sense to have separate `reason` (enum) and `message` (string) properties?

Reply:
@hunterwerlla / @rli  thoughts on this? This is the metrics data we'd send over on sam app creation

Reply:
I would opt for setting `metadata.result` to "Cancelled", and reason could be "User" or something similar